File: packages/typescript/tests/system/dragAndDrop.test.ts
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Drag and Drop", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { model, isMobile } = result;

      // Skip if using DeepSeek (no vision support yet)
      if (model.provider === "deepseek")
        skip("DeepSeek does not support vision yet");

      // Skip if using Appium driver (no drag and drop support in mobile browsers)
      if (isMobile)
        skip("Example doesn't support drag and drop in mobile browsers");

      return result;
    };
  });

  it("works", async ({ expect, setup }) => {
    const { al, $ } = await setup();

    await $.navigate("https://the-internet.herokuapp.com/drag_and_drop");

    const initialOrder = await al.get(
      "titles of squares ordered from left to right",
      { vision: true },
    );
    expect(initialOrder).toEqual(["A", "B"]);

    await al.do("move square A to square B");

    const finalOrder = await al.get(
      "titles of squares ordered from left to right",
      { vision: true },
    );
    expect(finalOrder).toEqual(["B", "A"]);
  });
});

File: packages/python/examples/pytest/drag_and_drop_test.py
from os import getenv

import pytest
from pytest import mark

from alumnium import Provider


@mark.xfail(
    getenv("ALUMNIUM_DRIVER", "selenium") == "appium-ios",
    reason="Example doesn't support drag and drop in mobile browsers",
)
def test_drag_and_drop(al, navigate):
    if al.model.provider == Provider.DEEPSEEK:
        pytest.xfail("No vision support yet")

    navigate("https://the-internet.herokuapp.com/drag_and_drop")
    assert al.get("titles of squares ordered from left to right", vision=True) == ["A", "B"]
    al.do("move square A to square B")
    assert al.get("titles of squares ordered from left to right", vision=True) == ["B", "A"]
